### ISSUE TITLE
[Question] Is it possible to return the final FileURI or the selected location URI

### ISSUE THREAD

**Issue by MarcooMateoo** [Amphiluke/cordova-plugin-save-dialog#1]
Is it possible to return the final FileURI or the selected location URI ?

**Comment by Amphiluke** [comment]
Hi,
Yes, it’s possible. Note however that this will be a security-scoped URL’s path, so you shouldn’t expect having file access by that URL. You will be able to use it for information purposes only.
The new minor version of the plugin with the proposed changes will be released soon.

**Comment by Amphiluke** [comment]
The plugin’s new version (1.1.1) is available on npm. The URI of the saved file is provided as a fulfillment value of the promise returned by the `saveFile` method (see [examples](https://github.com/Amphiluke/cordova-plugin-save-dialog#savefile) in README).

**Comment by MarcooMateoo** [comment]
Thanks @Amphiluke! :) 
we are actually planning on combining it using the cordova file plugin

**Comment by mreis1** [comment]
@Amphiluke I've the same question... 
I know the Uri is returned, but the Uri doesn't look like an "Absolute file Uri" that we can use in plugins such as [FileOpener2 ](https://github.com/pwlin/cordova-plugin-file-opener2)

Since I'm not that experience in Android internal API , I was trying to see how I could convert that File Descriptor Uri into an absolute Uri that I could later use to open the file. 

Uri being returned: `content://com.android.providers.downloads.documents/document/480`
Calling uri.getPath() outputs /document/488 but this still can't be used.

Do you have any tip on how to deal with this? 

Forgot to mention that I tried multiple approaches related such as https://stackoverflow.com/a/30827530/1084568   but still no luck.


FileOpen2 internally tries this:
```
try {
			CordovaResourceApi resourceApi = webView.getResourceApi();
			Uri fileUri = resourceApi.remapUri(Uri.parse(fileArg));
			fileName = fileUri.getPath();
		} catch (Exception e) {
			fileName = fileArg;
		}

File file = new File(fileName);
if (file.exists()) { // output false for /documents/480


```

**Comment by Amphiluke** [comment]
Hi,

There are working solutions on the Internet like [this one](https://stackoverflow.com/a/55352140). I haven’t tested it thoroughly but a quick experiment with the Android emulator gives promising results. I’ve pushed this [quick and dirty solution](https://github.com/Amphiluke/cordova-plugin-save-dialog/commit/6e8feb03ea734c5a32ad3b9701319f074e98c4be) to a temporal branch, [abs-url](https://github.com/Amphiluke/cordova-plugin-save-dialog/tree/abs-url). You may try installing the plugin from this branch and see if the solution works for you. But again: I haven’t tested it enough, so cannot guarantee it’s a “silver bullet”.

**Comment by MarcooMateoo** [comment]
Hi @mreis1,

we did use another [plugin ](https://github.com/vascoferreira/cordova-plugin-filepath-generic) in order to retrieve the file URI in android. 

Hope this helps you as well 
Cheers!

**Comment by mreis1** [comment]
Thank you for sharing @MarcooMateoo I'll give it a try. 
Also thank you  @Amphiluke for your commit, I'll try it tomorrow morning.

**Comment by mreis1** [comment]
Hello, @Amphiluke tried your experimental branch but FileUtil.getPath("content://com.android.providers.downloads.documents/document/496", ...) is outputting null. 

![image](https://user-images.githubusercontent.com/1719667/187156963-78ebac90-550d-4771-ba44-6505daf04135.png)

**Comment by mreis1** [comment]
@MarcooMateoo tried your suggestion but this lead me to a known issue reported [here](https://github.com/vascoferreira/cordova-plugin-filepath-generic/issues/1)

**Comment by Amphiluke** [comment]
Hello, @mreis1 

It appears that the problem can be solved using the approach described [here](https://stackoverflow.com/a/53021624) (i.e. iterating over all possible download path segments). But unfortunately there is another problem: accessing files in `all_downloads` requires the permission `android.permission.ACCESS_ALL_DOWNLOADS` which [can not be granted](https://stackoverflow.com/a/36798753) to a 3rd party app. Alas, I have no idea how to overcome this, sorry. Listing the permission in manifest and requesting the permission in runtime have no effect.

As of the current plugin, I just want to repeat: it provides the final URL for information purposes only. Access to a file saved by the user can be restricted by the system both in iOS and Android. If you manage to grant your app all required permissions and bypass the restriction, it will be awesome, but I believe this solution will be out of this plugin’s scope. :shrug:

**Comment by mreis1** [comment]
hello @Amphiluke , thank very much for the feedback and research in this subject matter. In fact your conclusions helped understanding what's possible to achieve and what's not. I understand why accessing downloads folder is a sensitive matter. In fact, most users have their files downloaded to this folder and some might contain sensitive data. Having apps accessing downloads folder in background could lead to data being stolen silently which ofc, could lead to privacy issues. 
Now, in my outsider point of view, I was expecting to be able  to open a file that I just wrote (for example, Android/iOS could generate a special file token (bound to the app that created the file) which could then be used to silently access the file we just created).  Sadly, this doesn't seem possible, but I'll try to dig a little bit more in this subject. 

If I get something, I'll come back to share ;-)
